DSharpPlus.Commands/Processors/TextCommands/Parsing/DefaultTextArgumentSplicer.cs

```diff
@@ -0,0 +1,147 @@
+namespace DSharpPlus.Commands.Processors.TextCommands.Parsing;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+public delegate string? TextArgumentSplicer(CommandsExtension extension, string text, ref int startAt);
+
+public class DefaultTextArgumentSplicer
+{
+    [Flags]
+    private enum State
+    {
+        None = 0,
+        Quoted = 1 << 0,
+        InlineCode = 1 << 1,
+        TripleCode = 1 << 2,
+        Escape = 1 << 3
+    }
+
+    public static string? Splice(CommandsExtension extension, string text, ref int startAt)
```

Comment:
all of this is not okay, and i'm not sure whether i like returning `string`s, either - that's an allocation for every parameter

Reply:
The design for splicing text arguments is more than encouraged to change. When opening the PR I actually forgot about this. Currently this implementation is a direct copy and paste from CommandsNext. I'll see if I can write up a better state machine.